lib/screens/product_detail.dart

```diff
@@ -586,6 +492,48 @@ class _ProductDetailScreenState extends State<ProductDetailScreen>
 
   Widget addToCartFAB() {
     return ScopedModelDescendant<MainModel>(
+      builder: (BuildContext context, Widget child, MainModel model) {
+        return _tabController.index == 0 ? FloatingActionButton(
+          child: Icon(
+            Icons.shopping_cart,
+            color: Colors.white,
+          ),
+          onPressed: () {
+            Scaffold.of(context).showSnackBar(processSnackbar);
+            selectedProduct.isOrderable
+                ? model.addProduct(
+                variantId: selectedProduct.id, quantity: quantity)
+                : null;
+            if (!model.isLoading) {
+              Scaffold.of(context).showSnackBar(completeSnackbar);
+            }
+          },
+          backgroundColor:
+          selectedProduct.isOrderable ? Colors.orange : Colors.grey,
+
+        ) : FloatingActionButton(
+
+            child: Icon(
+              Icons.add,
+              color: Colors.white,
+            ),
+            onPressed: () {
+              if(model.isAuthenticated) {
+                Navigator.of(context).push(MaterialPageRoute(
+                    builder: (context) => ReviewDetailScreen(selectedProduct)));
+              }
+              else
+              {
+                Scaffold.of(context).showSnackBar(LoginErroSnackbar);
```

Comment:
Please add an action button with label as 'Login' and Push the AuthenticationScreen as done for Add to Favorites IconButton.